kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -1227,18 +1213,41 @@ impl<'tcx> GotocCtx<'tcx> {
             loc,
         );
 
-        // Check that computing `bytes` would not overflow
+        // Check that computing `count` in bytes would not overflow
+        let (count_bytes, overflow_check) =
```

Comment:
This catches the `Rust` definition of UB, but will still cause problems in CBMC if we overflow the index bits, correct?

Reply:
Yes, and there is now a separate test for CBMC's wrap-around behavior.